### ISSUE TITLE
Depreciate Legacy Headers

### ISSUE THREAD

**Issue by 0xazure** [new-xkit/XKit#479]
In conjunction with #478, I would propose the depreciation of [Legacy Headers](https://github.com/new-xkit/XKit/wiki/Legacy-Headers) in favour of proper programmatic export of these values.  In essence, we would change the existing comment block, like below:

```js
//* TITLE Extension Title **//
//* VERSION 1.0.0 **//
//* DESCRIPTION A short description **//
//* DETAILS A much longer description should go here. **//
//* DEVELOPER new-xkit **//
//* FRAME false **//
//* BETA false **//
```

and convert it into CommonJS module exports, for example:

```js
var meta = {
  title = "Extension Title",
  version = "1.0.0",
  description = "A short description.",
  details = "A much longer description should go here.",
  developer = "new-xkit",
  frame = false,
  beta = false
};

module.exports.meta = Object.freeze(meta);
```

**Comment by Wolvan** [comment]
:+1:

**Comment by bvtsang** [comment]
We'll also need to update the following wiki pages:

- https://github.com/new-xkit/XKit/wiki/Example-Extension
- https://github.com/new-xkit/XKit/wiki/Legacy-Headers
- https://github.com/new-xkit/XKit/wiki/Writing-a-New-Extension
- possibly others

**Comment by Wolvan** [comment]
First we have to phase the old headers out, @bvtsang

**Comment by bvtsang** [comment]
Right, I'm just mentioning the wiki pages because we'll need to make sure the deprecation of the headers is consistent with the information we have on the wikis. We don't want people getting conflicting information.

Also, mentioning the wiki stuff here helps us track it.
